tests/rptest/services/kerberos.py

```diff
@@ -495,3 +495,99 @@ def start_node(self, node, **kwargs):
     def _service_principal(self, node, primary: str = "redpanda"):
         fqdn = RedpandaService.get_node_fqdn(node)
         return f"{primary}/{fqdn}@{self.realm}"
+
+
+class ActiveDirectoryKdc:
+    def __init__(self, logger, remote_account: RemoteAccount, realm: str,
+                 keytab_password: str, upn_user: str, spn_user: str):
+        self.logger = logger
+        self.remote_account = remote_account
+        self.realm = realm
+        self.keytab_password = keytab_password
+        self.upn_user = upn_user
+        self.spn_user = spn_user
+        self.nodes = [
+            namedtuple("node", "name account")(remote_account.hostname,
+                                               remote_account)
+        ]
+
+    def ssh_input_output(self,
+                         cmd,
+                         allow_fail=False,
+                         combine_stderr=True,
+                         timeout_sec=None,
+                         in_lines=None):
+        """Runs the command via SSH and captures the output, returning it as a string.
+
+        :param cmd: The remote ssh command.
+        :param allow_fail: If True, ignore nonzero exit status of the remote command,
+               else raise an ``RemoteCommandError``
+        :param combine_stderr: If True, return output from both stderr and stdout of the remote process.
+        :param timeout_sec: Set timeout on blocking reads/writes. Default None. For more details see
+            http://docs.paramiko.org/en/2.0/api/channel.html#paramiko.channel.Channel.settimeout
+        :param in_lines: An iterable of strings to pass to stdin, each string is flushed, newlines are not added.
```

Comment:
Does `stdin.write()` flush with each write?  Because it appears you write each line and then flush after each line is written

Reply:
I changed the implementation, but not the comment.